Crash: assign boolean or result to float

```c
void foo(void) {
    float f = 2.0 || 0;
}
```

```
panic: access of union field 'int' while field 'float' is active
/Users/ehaas/source/arocc/src/aro/Value.zig:205:46: 0x103d44402 in intToFloat (arocc)
    return switch (comp.interner.get(v.ref()).int) {
                                             ^
/Users/ehaas/source/arocc/src/aro/Parser.zig:5589:35: 0x103d15b0f in floatCast (arocc)
            try res.val.intToFloat(float_ty, p.comp);
                                  ^
/Users/ehaas/source/arocc/src/aro/Parser.zig:6086:34: 0x103ca033c in coerceExtra (arocc)
                try res.floatCast(p, unqual_ty);
                                 ^
```

--- a/src/aro/Parser.zig
+++ b/src/aro/Parser.zig
@@ -6415,6 +6415,8 @@ fn lorExpr(p: *Parser) Error!Result {
         if (try lhs.adjustTypes(tok, &rhs, p, .boolean_logic)) {
             const res = lhs.val.toBool(p.comp) or rhs.val.toBool(p.comp);
             lhs.val = Value.fromBool(res);
+        } else {
+            lhs.val.boolCast(p.comp);
         }
         try lhs.boolRes(p, .bool_or_expr, rhs);
     }

--- a/test/cases/float values.c
+++ b/test/cases/float values.c
@@ -1,2 +1,6 @@
 _Static_assert(-1.0 - 1.0 == -2.0, "");
 _Static_assert(-2.0f == -2.0, "");
+_Static_assert(1.0 == (2.0||0), "");
+void foo(void) {
+    float f = 2.0 || 0;
+}

